.travis.yml

```diff
@@ -5,6 +5,7 @@ os:
   - osx
 
 dist: bionic
+osx_image: xcode11.3
```

Comment:
It was explicitly removed to have less maintenance.
Anyway you should install the latest version from brew, not by bumping this.

Reply:
@Chocobo1, can you explain these requirements for me (I don't know much about all this TravisCI stuff)?

Reply:
Basically travisci will bump the default xcode image when they see necessary: https://blog.travis-ci.com/2018-07-19-xcode9-4-default-announce
And since the current default (when we don't specify one) xcode 9.4 image mostly fits our usage, I don't see any reason to touch it, mostly to have less maintenance as I've said before.

The only problem here is xcode 9.4 image ships with older cmake, but this is not a problem because [Homebrew](https://en.wikipedia.org/wiki/Homebrew_(package_manager)) exists and we are already using it.
Homebrew provides up-to-date packages: https://formulae.brew.sh/formula/cmake
So that's why I suggested the change.

Reply:
@Chocobo1
Am I missing something? If your "less maintenance" argument stands entirely on "not having to maintain this extra line", then I disagree with it. We should explicitly set the version that suits our needs, not rely on whatever Travis decides to set as default.

Reply:
>We should explicitly set the version that suits our needs, not rely on whatever Travis decides to set as default.

Still I don't see why the default xcode 9.4 isn't suitable.
Also your logic can expand to C++ compiler as well and all other things. Do you want to make all of them explicit in our CI scripts, by explicitly installing the "specific" versions you want?

>Besides, installing a certain CMake version from Homebrew are also 1 or more lines we have to "maintain".

That is not really true, there is no "certain CMake version", it is always the latest version when the command is `brew install cmake`.

---

I rarely got positive results/feedback reviewing your PR and I deeply regret stepping into this one, I'll see my way out.

Reply:
Anyway it's a good point to always use minimum supported versions in CI environments otherwise we may accidentally break compatibility.

Reply:
@Chocobo1
>Still I don't see why the default xcode 9.4 isn't suitable.

It's not even the toolchain used to build official releases anymore.

>Also your logic can expand to C++ compiler as well and all other things. Do you want to make all of them explicit in our CI scripts, by explicitly installing the "specific" versions you want?

What an obtuse slippery slop fallacy. Of course there are some things that worth explicitly specifying the version of, while the same might not be true for others.

> That is not really true, there is no "certain CMake version", it is always the latest version when the command is brew install cmake.

That's true, but it's another line to maintain - oh no!

To be clear - I think both lines should be added - because it doesn't make sense for the CMake version to be tied to the rest of the Apple toolchain when it can just be the latest. I'm just saying that while we're at it, we can bump the toolchain version, and explicitly state what it should be.

> I rarely got positive results/feedback reviewing your PR and I deeply regret stepping into this one, I'll see my way out.

You're not even trying to collaborate. Why are you so against discussion? It's strange to me, coming from a person who started a discussion because of one extra line that's actually useful, but at the same time is perfectly indifferent with hundreds of lines of qmake/autools bloat in the code base.

I've always listened fairly to your opinions and criticisms. But you decided to start acting like a drama queen since some time ago and claim I am the source of the problem. Well, how about you talk about it? I'm all ears. Instead, you block me (yes, I know you did this a some weeks back). Really mature of you.

Reply:
Wow, getting real personal here and starting some attacks...

>You're not even trying to collaborate. 

Why would I need to do that, I can ignore anything that doesn't interests me, can't I?

>You're not even trying to collaborate....Well, how about you talk about it? 

From my viewpoint, you're the one that doesn't take concern from other voices (at least mine) and only fixed on pushing your agenda.

>Instead, you block me (yes, I know you did this a some weeks back). 

Since you asked, as a start, I would appreciate you stop looping/ping me in into every discussion thread...

>Really mature of you.

Good luck!

Reply:
Guys, please cool down and let's solve the problem. You don't let we go any further...
As I said above, the best option is to use the minimum supported versions here (by the way, this was an approved policy earlier, and I don't see any objective point in changing this, otherwise CI will become almost useless).
So the question is what ways we can do this, and what are the objective pros and cons of each way (the number of configuration lines does not count if the difference is not measured in tens/thousands).

Reply:
@glassez 
As far as the toolchain goes, I don't see why we should use a version lower than what is used for the official builds. That's all. As it happens, the bundled version of CMake that comes in the most recent toolchain is sufficient to fix the current problem, but I also agree that installing the latest one from brew is the best option.

@Chocobo1
> Why would I need to do that, I can ignore anything that doesn't interest me, can't I?

You can, but not by claiming that others are "hand-waving" your concerns when that's not what's happening.

> I tried my best, but I sense from your reply that you won't even give a second to thought about it nor at least try it, just hand waiving it away...

Prime example. I asked if I was missing something. I presented my argument based on what I understood, and waited for feedback. You reply, but gaslight me by posting stuff like this, as if my post was in anyway not collaborative:

>I rarely got positive results/feedback reviewing your PR and I deeply regret stepping into this one, I'll see my way out.

And again:

>Again, you put the responsibility of a group decision onto my back just to attack me.

I did not put the responsibility of a group decision onto your back. I never claimed nor implied it was _only your decision_, just that you agree with it/are fine with it, and are one if its supporters. Is that not true? Given that, I'm just stating how inconsistent that is with your current position in regards to this one line change.

>You only claim to do so.

>From my viewpoint, you're the one that doesn't take concern from other voices (at least mine) and only fixed on pushing your agenda.

This is flat out wrong. Feel free to look at any of my PRs and contributions. I'm interested in pushing what's best for the project. Whenever there was consensus contrary to my initial beliefs, or whenever it turned out that I was wrong, I've always corrected whatever needed to be corrected. I even make a point to use strike-through and not delete stuff I've said previously, even if I've changed my mind, in the interest of transparency and accountability. I've always listened to feedback and implemented requested changes whenever they turned out to be the right thing to do. I've had multiple exchanges with many people here, and it's just you that has a problem with me. 

With anybody else, we've always managed to get to the bottom of the issue at hand without one party childishly abandoning the discussion and claiming the other was "hand-waving" concerns away.
Yes, I like to be thorough in my discussions. That can mean they take longer. If that is a problem, you could just politely state "sorry, I must focus on other things, so I might take a little while longer, or ask someone else about XYZ".

> I would appreciate you stop looping/ping/addressing me in into every discussion thread... My email box was flooded with topics I'm uninterested in, thanks to you.

You are one of the biggest and longest-time contributors to the project. It is only natural people ping you for advice/thoughts/executive decisions. There is a certain responsibility in being one of the biggest fish. Also, IMO "mailbox flood" is not a good reason to block someone. Categorization and filtering in mailboxes is a solved problem. But, even assuming it's not, I hope you realize the correct thing to do is to post something in the team page "hey, I'd appreciate if you/others didn't tag me so much for the next few days/weeks, I'm busy doing other stuff" or "Hey Francisco, I should let you know I've temporarily, because my inbox is too flooded with your pings and currently my mailbox organization is not so good. For urgent stuff you can make a team mention, etc etc etc". Blocking someone without explanation or recourse is not nice and not something you do in a team. Or hell, just flat out say "hey, please don't tag me for these kinds of issues: webui, crashes, etc, etc etc, (just examples) as I'm not interested in that".

The way I see it, there were much better ways you could have handled this.

I've said my piece. I'm always interested in moving forward and working together more harmoniously in the future. Just please communicate with me in the team page, don't block without saying anything. This goes for anyone else reading this. If anyone has a problem with the way me or anyone else does things, speak out and address the problems. It can always be worked out without too much fuss if is handled early and with good communication.

Reply:
Just found another reason to change the macOS CI toolchain over to XCode 11: https://doc.qt.io/qt-5/macos.html - checkout the "Build environment" column in the "Supported versions" section, right at the top of the page - it's XCode 11.

Reply:
>>Again, you put the responsibility of a group decision onto my back just to attack me.
>
>I did not put the responsibility of a group decision onto your back. I never claimed nor implied it was only your decision, just that you agree with it/are fine with it, and are one if its supporters. 

I never understand why you need to clarify this, by making it looks like non-hostile. The way I see it, it is malicious, why don't you challenge that on others, why only me?

>Is that not true? 

Furthermore, what that has to do with the topic at hand? 

>It is only natural people ping you for advice/thoughts/executive decisions.

Then I'm just doing my job in this very PR, I insisted on something, see how it turns out...

>There is a certain responsibility in being one of the biggest fish. 

Sorry, I don't wish to have it. I'm only interested in some of the topics, not every one.

>But, even assuming it's not, I hope you realize the correct thing to do is to post something in the team page 

Thanks, I'll keep that in mind. A related thing is I wish you stop using `@` prefix to address me, it forcefully subscribe me back into a thread.

>I'm always interested in moving forward and working together more harmoniously in the future. 

Me too, but when there will be different opinion

Reply:
>Just found another reason to change the macOS CI toolchain over to XCode 11:

We support Qt 5.9 as the minimum, here is the doc although it doesn't mention XCode version.
https://doc-snapshots.qt.io/qt5-5.9/supported-platforms.html

Reply:
>We support Qt 5.9 as the minimum, here is the doc although it doesn't mention XCode version.
https://doc-snapshots.qt.io/qt5-5.9/supported-platforms.html

I thought support for Qt versions that old was only relevant for (stable) Linux distributions such as Ubuntu/Debian. Windows and macOS releases just get built with whatever the most recent version is (or at least as recent as possible).

Plus, correct me if I'm wrong, but I am quite sure I've seen the maintainer mention in another thread that we no longer officially support macOS <= 10.12 anyway. Not to mention that such versions have already been EOL'd by Apple, and on the macOS world, fortunately, things move much faster (a large percentage getting "stuck" on an obsolete version doesn't really happen, as opposed to the Windows XP/7 fenomenon).

Reply:
>As far as the toolchain goes, I don't see why we should use a version lower than what is used for the official builds.

>Just found another reason to change the macOS CI toolchain over to XCode 11

I'm sorry, whatever your reasons, there can't be any valid reason to change supported tools (and versions) by accident (in some regular PR). We usually devote a separate discussion to this issue (for example, before a major update). This case with CMake is an exception, since it is not the official build system for the project (but I'm already starting to regret that we made this exception; if there is no consensus here, it's easier to undo previous changes). But we should not change any other tools here.
As for CMake... We decided to upgrade the version to 3.16, so it's also a very bad idea to use a newer one here.

And Yes, I still don't agree with this slogan that "we should always use only the newest version of CMake, and not support the older ones", and I am against it, at least until it is collegially approved as the official policy of the project.

So let's stop all subjective quarrels and start solving the problem based on the current rules of the project.

Reply:
If Linux part of this PR has no objections I suggest to merge it separately. @Chocobo1?

Reply:
When I say "we should always use only the newest version of CMake, and not support the older ones", I only mean the runtime version, not what we set as the minimum supported version in the build script, which I also agreed could be 3.16.

Reply:
Note: I agree at this point the toolchain change should be handled in another PR, but according to the build logs, travis is still using the CMake version from the toolchain (3.17.0), not the one installed from the homebrew package (which should be 3.17.2). So I'm concerned the build will break if we revert the toolchain now (because I suspect that travis will not find and use automatically the homebrew-installed version).

Maybe reverting the toolchain change should be handled in another PR instead, to keep the build working for now? Unless we can quickly figure out how to get travis to use the homebrew supplied version. I'm investigating that now.

Reply:
>If Linux part of this PR has no objections I suggest to merge it separately. @Chocobo1?

OK.

Reply:
@glassez all is fixed, commits cleaned up and toolchain change removed to be solved in another PR.